DeepSpeech.py

```diff
@@ -1444,6 +1445,11 @@ def end(self, session):
     if FLAGS.summary_secs > 0:
         hooks.append(tf.train.SummarySaverHook(save_secs=FLAGS.summary_secs, output_dir=FLAGS.summary_dir, summary_op=merge_all_summaries_op))
 
+    # Hook wih number of checkpoint files to save in checkpoint_dir
+    if FLAGS.max_to_keep > 0:
+	saver_var = tf.train.Saver(max_to_keep=FLAGS.max_to_keep)
+        hooks.append(tf.train.CheckpointSaverHook(checkpoint_dir=FLAGS.checkpoint_dir, save_secs=FLAGS.checkpoint_secs, saver=saver_var))
```

Comment:
@tilmankamp: I am not sure if I get it. If we remove the CheckpointSaverHook from MonitoredTrainingSession. How would it limit the number of checkpoint files? MonitoredTrainingSession can have multiple hooks passed as an argument.